pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -75,10 +79,12 @@ func NewNamespaceController(kubeClient kube.Client, caBundleWatcher *keycertbund
 		ObjectFilter:  c.GetFilter(),
 	})
 	c.namespaces = kclient.New[*v1.Namespace](kubeClient)
+	// kube-system is not skipped to enable deploying ztunnel in that namespace
```

Comment:
I think in InsertDataToConfigMap we will want to add `errors.IsUnauthorized()` to the error handling and log at info level (something like "Skipping kube-system due to lack of permission") without retries?

Otherwise we are going to retry 5 times ( and could retrigger if the namespace changes) with a concerning-looking error message?

Reply:
Let me spin up a cluster that denies kube-system so I can make this more concrete maybe (and make sure IsUnauthorized is correct)

Reply:
actually its IsForbidden we want I think (at least on GKE...):

```
			if errors.IsForbidden(err) {
				log.Infof("skip writing ConfigMap %v/%v as we do not have permissions to do so", meta.Namespace, meta.Name)
				return nil
			}
```

Reply:
Thanks for suggestion. I simulated the forbidden issue on kind with this [patch](https://github.com/jewertow/upstream-istio/commit/2404d150d61e1de45861b1cabd50e91a1c6a8282) and it works as expected.

Before:
```
2024-02-08T20:23:51.593609Z     error   controllers     error handling kube-system, retrying (retry count: 1): error when creating configmap istio-ca-root-cert: configmaps is forbidden: User "system:serviceaccount:istio-system:istiod" cannot create resource "configmaps" in API group "" in the namespace "kube-system"   controller=namespace controller
...
```
After:
```
2024-02-08T20:20:22.403813Z     info    skip writing ConfigMap kube-system/istio-ca-root-cert as we do not have permissions to do so
```